$NetBSD$

Retain the pkgsrc working-grep fix and add an opt-in checked base gettext
path for native OS consumers. EmberBSD, AI-assisted; not yet upstream.

--- configure.orig	2026-10-28 21:34:07
+++ configure	2026-10-28 21:34:07
@@ -3017,6 +3017,7 @@
 am__quote'
 ac_subst_files=''
 ac_user_opts='
+with_base_libintl
 enable_option_checking
 enable_silent_rules
 enable_dependency_tracking
@@ -3725,6 +3726,7 @@
                           Packager info for bug reports (URL/e-mail/...)
   --with-libiconv-prefix[=DIR]  search for libiconv in DIR/include and DIR/lib
   --without-libiconv-prefix     don't search for libiconv in includedir and libdir
+  --with-base-libintl     use a checked base libintl
   --with-libintl-prefix[=DIR]  search for libintl in DIR/include and DIR/lib
   --without-libintl-prefix     don't search for libintl in includedir and libdir
 
@@ -5462,14 +5464,6 @@
 
 
 
-if test -n "$GREP" || test -n "$EGREP"; then
-  as_fn_error $? "no working 'grep' found
-  A working 'grep' command is needed to build GNU Grep.
-  This 'grep' should support -e and long lines.
-  On Solaris 10, install the package SUNWggrp or SUNWxcu4.
-  On Solaris 11, install the package text/gnu-grep or system/xopen/xcu4." "$LINENO" 5
-fi
-
 
 
 
@@ -61092,6 +61086,48 @@
 
 
     INTLLIBS="$LIBINTL"
+
+# An explicitly selected base libintl is checked via its public API.
+case ${with_base_libintl-no} in
+  yes|no) ;;
+  *) as_fn_error $? "invalid --with-base-libintl value" "$LINENO" 5 ;;
+esac
+if test "${with_base_libintl-no}" = yes; then
+  if test "${enable_nls-yes}" = no; then
+    as_fn_error $? "base libintl requires NLS" "$LINENO" 5
+  fi
+  saved_LIBS=$LIBS
+  LIBS="-lintl $LIBS"
+  { printf '%s\n' "$as_me:${as_lineno-$LINENO}: checking for the base libintl public API" >&5
+    printf %s "checking for the base libintl public API... " >&6; }
+  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+#include <libintl.h>
+#include <locale.h>
+int main (void) {
+  bindtextdomain("grep", "/usr/share/locale");
+  bind_textdomain_codeset("grep", "UTF-8");
+  textdomain("grep");
+  return !gettext("x") || !dgettext("grep", "x") ||
+    !dcgettext("grep", "x", LC_MESSAGES) || !ngettext("x", "xs", 2) ||
+    !dngettext("grep", "x", "xs", 2) ||
+    !dcngettext("grep", "x", "xs", 2, LC_MESSAGES);
+}
+_ACEOF
+  if ac_fn_c_try_link "$LINENO"; then
+    { printf '%s\n' "$as_me:${as_lineno-$LINENO}: result: yes" >&5
+      printf '%s\n' yes >&6; }
+  else
+    as_fn_error $? "base libintl public API link failed" "$LINENO" 5
+  fi
+  rm -f core conftest.err conftest.$ac_objext conftest.beam conftest$ac_exeext conftest.$ac_ext
+  LIBS=$saved_LIBS
+  USE_NLS=yes
+  POSUB=po
+  LIBINTL=-lintl
+  LTLIBINTL=-lintl
+  INTLLIBS=-lintl
+  printf '%s\n' '#define ENABLE_NLS 1' '#define HAVE_GETTEXT 1' '#define HAVE_DCGETTEXT 1' >>confdefs.h
+fi
 
 
 
